helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/CodeTransformations.java

```diff
@@ -304,6 +314,59 @@ public static String transformParseCalls(String code, AvroVersion minSupportedVe
     return prefix + newParseCall + restOfCode;
   }
 
+  /**
+   * escapes any control characters inside string literals that are part of an argument json
+   * PACKAGE PRIVATE FOR TESTING
+   * @param str a piece of json, as a java string literal
+   * @return the given piece of json, safe for use as a java string literal
+   */
+  static String escapeJavaLiteral(String str) {
+    if (str == null || str.isEmpty()) {
+      return str;
+    }
+
+    int quotedStrStart; //start of current quoted string (index of the starting quote character)
+    int quotedStrEnd;   //end of current quoted string (index of the end quote character)
+    int lastEnd = 0;    //end of previous quoted string (index of the end quote character)
+    StringBuilder result = new StringBuilder(str.length());
+
+    quotedStrStart = str.indexOf("\\\"", lastEnd);
+    while (quotedStrStart >= 0) {
+      quotedStrEnd = quotedStrStart;
+      while (true) {
```

Comment:
This nested loop is confusing to me... I guess you're looking for the "regular" JSON strings in the schema, and then escaping the interior of those strings? It's great that you sprinkled a good amount of comments throughout, but I think it might help to add one more at the beginning here to describe the flow a bit more.